miner/worker.go

```diff
@@ -978,6 +978,12 @@ loop:
 			break
 		}
 		// If we have collected enough transactions then we're done
+		// Originally we only limit l2txs count, but now strictly limit total txs number. 
+		if !w.chainConfig.Scroll.IsValidL2TxCount(w.current.tcount+1) {
```

Comment:
```suggestion
		if !w.chainConfig.Scroll.IsValidTxCount(w.current.tcount+1) {
```

Reply:
and we can remove `!w.chainConfig.Scroll.IsValidL2TxCount(l2TxCount)` below, and add it back if necessary.